src/domain/messages/messages.repository.ts

```diff
@@ -52,28 +57,38 @@ export class MessagesRepository implements IMessagesRepository {
     signature: string;
     origin: string | null;
   }): Promise<unknown> {
+    const safe = await this.safeRepository.getSafe({
+      chainId: args.chainId,
+      address: args.safeAddress,
+    });
     const transactionService = await this.transactionApiManager.getApi(
       args.chainId,
     );
-
-    return transactionService.postMessage({
+    const response = await transactionService.postMessage({
       safeAddress: args.safeAddress,
       message: args.message,
       safeAppId: args.safeAppId,
       signature: args.signature,
       origin: args.origin,
     });
+    const message = MessageSchema.parse(response);
+    this.messageVerifier.verifyMessageHash({
```

Comment:
We should verify before the `POST` otherwise the check is worthless.

Reply:
My intention was to check the hash the Transaction Service returns is consistent with the payload sent. Does this make sense to you? Otherwise which hash would we check? The user request does not include the hash if I'm not wrong.